src/qinfer/hyper_heuristic_optimisers.py

```diff
@@ -0,0 +1,202 @@
+        self._fitness[0]["velocities"] = self.update_velocities(
+                                                self._fitness[0]["params"], 
+                                                self._fitness[0]["velocities"], 
+                                                local_attractors["params"],
+                                                global_attractor["params"],
+                                                omega_v, phi_p, phi_g)
+
+        for itr in range(1, n_pso_iterations):
+            #Update the particle positions
+            self._fitness[itr]["params"] = self.update_positions(
+                self._fitness[itr - 1]["params"], 
+                self._fitness[itr - 1]["velocities"])
+
+            # Apply the boundary conditions if any exist
+            if self._boundary_map is not None:
+                self._fitness[itr]["params"] = self._boundary_map(self._fitness[itr]["params"])
+
+            # Recalculate the fitness function
+            self._fitness[itr]["fitness"] = self.evaluate_fitness(
+                self._fitness[itr]["params"],
+                serial_map=serial_map)
+
+            # Find the new attractors
+            local_attractors, global_attractor = self.update_attractors(
+                self._fitness[itr], 
+                local_attractors, 
+                global_attractor)
+
+            # Update the velocities
+            self._fitness[itr]["velocities"] = self.update_velocities(
+                self._fitness[itr]["params"], 
+                self._fitness[itr - 1]["velocities"], 
+                local_attractors["params"],
+                global_attractor["params"],
+                omega_v, phi_p, phi_g)
+
+        return global_attractor
+
+    def evaluate_fitness(self, particles, serial_map):
+        fitness_function = partial(self.fitness_function)
+        fitness = np.empty([len(particles)], dtype=np.float64)
+        fitness = serial_map(self.fitness_function, particles)
+        return fitness
+        
+    def update_positions(self, positions, velocities):
+        updated = positions + velocities
+        return updated
+
+    def update_velocities(self, positions, velocities, local_attractors, global_attractor, omega_v, phi_p, phi_g):
+        random_p = np.random.random_sample(positions.shape)
+        random_g = np.random.random_sample(positions.shape)
+        updated = omega_v * velocities + phi_p * random_p * (local_attractors - positions) + phi_g * random_g * (global_attractor - positions) 
+        return updated
+
+    def update_attractors(self, particles, local_attractors, global_attractor):
+        for idx, particle in enumerate(particles):
+            if particle["fitness"] < local_attractors[idx]["fitness"]:
+                local_attractors[idx] = particle
+        global_attractor = local_attractors[np.argmin(local_attractors["fitness"])]
+        return local_attractors, global_attractor
+    
+class PerfTestMultipleAbstractor:
```

Comment:
I think the term ``Abstractor`` here doesn't quite communicate what the class is doing. I'd suggest something like ``PerformanceObjective``, perhaps? Also, this should inherit from ``object`` so that it's forced to be a new-style class even on Python 2.